internal/client/ierror.go

```diff
@@ -0,0 +1,15 @@
+package client
+
+import "fmt"
+
+type iError struct {
+	status string
+}
+
+func newIError(status string) *iError {
+	return &iError{status: status}
+}
+
+func (i iError) Error() string {
```

Comment:
must be `func (i *iError) Error() string {`, otherwise new instance will be created every times this method is called